stub/src/main/java/io/grpc/stub/Calls.java

```diff
@@ -188,7 +190,7 @@
   public static <ReqT, RespT> RespT blockingClientStreamingCall(
       Call<ReqT, RespT> call,
       Iterator<ReqT> clientStream) {
-    SettableFuture<RespT> responseFuture = SettableFuture.create();
+    GrpcFuture<RespT> responseFuture = new GrpcFuture<RespT>(call);
```

Comment:
Bah, you're right. And that is also why you had to redeclare set and setException in order to give visibility to UnaryStreamToFuture.

LGTM